core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1982,6 +1978,8 @@ private[spark] class DAGScheduler(
               case failedMapStage: ShuffleMapStage =>
                 // Mark all the map as broken in the map stage, to ensure retry all the tasks on
                 // resubmitted stage attempt.
+                // TODO: SPARK-32923: Clean all push-based shuffle metadata like merge enabled and
+                // TODO: finalized as we are clearing all the merge results.
```

Comment:
Nit: this TODO should be moved to inside `submitMissingTasks` (line 1326) where it's handling indeterminate stage.